src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobServiceClient;
+import uk.gov.hmcts.reform.blobrouter.data.model.Envelope;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class DuplicateFinder {
+
+    private final BlobServiceClient storageClient;
+    private final EnvelopeService envelopeService;
+
+    public DuplicateFinder(
+        BlobServiceClient storageClient,
+        EnvelopeService envelopeService
+    ) {
+        this.storageClient = storageClient;
+        this.envelopeService = envelopeService;
+    }
+
+    public List<Envelope> findIn(String containerName) {
```

Comment:
may be worth making it more specific ? `findEnvelopesMarkedDeleted` or something similar ?

Reply:
it's finding duplicates

Reply:
it'd finding duplicates in given container. Notice it iterates over blobs.

Reply:
Changed return type to `BlobItem` for clarity (all we actually need is file 'identifier' (name + container), but `Envelope` might have been misleading)

Reply:
(I may add a `Duplicate` model later if it turns out we need the envelope ID)